interactive/dashboard/api.py

```diff
@@ -0,0 +1,1036 @@
+        }
+        log("returning basic info:", context)
+        return JsonResponse(context)
+
+
+class GetFlow(views.APIView):
+    def get(self, request):
+        log("****** GET request received GET_FLOW ******")
+        if not request.session.session_key:
+            request.session.create()
+        csv_results_path = os.path.join(
+            os.environ['MUSIC_BOX_BUILD_DIR'],
+            request.session.session_key+"/output.csv")
+        log("fetching flow from: "+csv_results_path)
+        log("using data:", request.GET.dict())
+        csv_results_path = os.path.join(
+            os.environ['MUSIC_BOX_BUILD_DIR'],
+            request.session.session_key+"/output.csv")
+        react_path = (os.path.join(
+            settings.BASE_DIR, 'configs/'+request.session.session_key)
+            +"/camp_data/reactions.json")
+        path_to_template = os.path.join(
+            settings.BASE_DIR,
+            "dashboard/templates/network_plot/flow_plot.html")
+        flow = create_and_return_flow_diagram(request.GET.dict(),
+                                              react_path, path_to_template,
+                                              csv_results_path)
+        return HttpResponse(flow)
+
+
+class DownloadConfig(views.APIView):
+    def get(self, request):
+        log("****** GET request received DOWNLOAD_CONFIG ******")
+        if not request.session.session_key:
+            request.session.create()
+        log("session key: "+request.session.session_key)
+        destination_path = os.path.join(
+            settings.BASE_DIR,
+            "dashboard/static/zip/"
+            +request.session.session_key
+            +"/config_copy")
+        zip_path = os.path.join(
+            settings.BASE_DIR, "dashboard/static/zip/output/"
+            +request.session.session_key
+            +"/config.zip")
+        conf_path = os.path.join(
+            settings.BASE_DIR, 'configs/'+request.session.session_key)
+        log("destination path: "+destination_path)
+        log("zip path: "+zip_path)
+        log("conf path: "+conf_path)
+        create_config_zip(destination_path, zip_path, conf_path)
+
+        fl_path = zip_path
+        zip_file = open(fl_path, 'rb')
+        response = HttpResponse(zip_file, content_type='application/zip')
+        response['Content-Disposition'] = 'attachment; filename="%s"' % 'config.zip'
+        return response
+
+
+class DownloadResults(views.APIView):
+    def get(self, request):
+        log("****** GET request received DOWNLOAD_RESULTS ******")
+        if not request.session.session_key:
+            request.session.create()
+        log("session key: "+request.session.session_key)
+        fl_path = os.path.join(
+            os.environ['MUSIC_BOX_BUILD_DIR'],
+            request.session.session_key+"/output.csv")
+        now = datetime.now()
+        filename = str(now) + '_model_output.csv'
+
+        fl = open(fl_path, 'r')
```

Comment:
Not sure how much it betters, but maybe we should use something like the example [here](https://stackoverflow.com/a/36394206/5217293) so that we don't leave the file open

Reply:
It looks like this `fl` file handle is no longer used. Am I missing something here?